File: frontend/src/types/book.ts
export interface Book {
  bookId: number
  title: string
  author: string
  publisher: string
  isbn: string
  classification: string
  category: string
  pageCount: number
  price: number
}

/** Payload for create/update (no id). */
export interface BookInput {
  title: string
  author: string
  publisher: string
  isbn: string
  classification: string
  category: string
  pageCount: number
  price: number
}

export interface PagedBooksResponse {
  books: Book[]
  totalCount: number
  page: number
  pageSize: number
}

export interface CartItem {
  bookId: number
  title: string
  price: number
  quantity: number
}

File: backend/Mission11.api/Models/Book.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Mission11.api.Models;

[Table("Books")]
public class Book
{
    [Key]
    [Column("BookID")]
    public int BookId { get; set; }

    [Column("Title")]
    public string Title { get; set; } = string.Empty;

    [Column("Author")]
    public string Author { get; set; } = string.Empty;

    [Column("Publisher")]
    public string Publisher { get; set; } = string.Empty;

    [Column("ISBN")]
    public string Isbn { get; set; } = string.Empty;

    [Column("Classification")]
    public string Classification { get; set; } = string.Empty;

    [Column("Category")]
    public string Category { get; set; } = string.Empty;

    [Column("PageCount")]
    public int PageCount { get; set; }

    [Column("Price")]
    public double Price { get; set; }
}
